lesson-003-web-server-graceful-shutdown/README.md

```diff
@@ -0,0 +1,104 @@
+# Lesson 3: Web server graceful shutdown
+
+Taking as a foundation the webserver created in [lesson 000](../lesson-000-web-server/), we will be adding `graceful shutdown` functionality to it.
+
+In order to achieve this, we will be using `channels` and `goroutines`
+
+## Knowledge bits
+
+
+### Goroutines
+
+A `goroutine` is a lightweight thread of execution managed by the Go runtime; it is the way we have to run a piece of code concurrently with the original calling code, as we can see explained in the [oficial documentation.](https://tour.golang.org/concurrency/1)
+
+A small example could be:
+
+```
+func say(s string) {
+	for i := 0; i < 5; i++ {
+		time.Sleep(100 * time.Millisecond)
+		fmt.Println(s)
+	}
+}
+
+func main() {
+	go say("k8s")
+	say("contribute")
+}
+```
+
+Calling `go printMyString()` is where we start a new `goroutine`, calling our already defined `printMyString` function, while the `main` function runs in its own `goroutine` (called the _main goroutine_)
```

Comment:
Need to update the name of the function from `printMyString` to `say`.

Reply:
🤦